src/redis-cli.c

```diff
@@ -293,6 +293,9 @@ static long getLongInfoField(char *info, char *field);
 /*------------------------------------------------------------------------------
  * Utility functions
  *--------------------------------------------------------------------------- */
+#if !defined strlcpy
+size_t strlcpy(char *dst, const char *src, size_t dsize);
+#endif
```

Comment:
i don't like this thing at all.
i.e. creating a forward declaration if a define doesn't exist...
why would it ever exist? does the C standard normally declares it as a preprocessor definition?

Reply:
strlcpy is apart of some standard C libraries, and not apart of others. Notably it's available on Macs running 10.13 and higher.

Reply:
Is it really defined as a preprocessor statement?

Reply:
@oranagra this was only added since is defined in Darwin. I agree this is not optimal, but with Darwin there are some issues due to the fact that it  uses macros to redefine some known function.

Reply:
so maybe the better solution is to have the preprocessor detect darwin and un-define it's defines?
it's really odd to me to declare and implement a function based on whether or not it is defined as a preprocessor.
and i also don't like the fact that we sometimes use our own implementation and other times we use the systems's.
so choosing a different name, or undefining what darwin defines, could maybe solve that better.

Reply:
I don't think we should be checking for darwin, because it's not a darwin thing. I think the option of always using redis_strlcpy makes more sense, but I also think it's perfectly okay to use a system provided implementation if it's available on the system.